src/common/base/src/tls.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::io::{BufReader, ErrorKind};
+
+use rustls::{Certificate, PrivateKey, ServerConfig};
+use rustls_pemfile::{certs, pkcs8_private_keys};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Default, Serialize, Deserialize, Clone)]
+pub struct TlsOption {
+    pub cert_path: String,
+    pub key_path: String,
```

Comment:
Instead of requiring `Option<TlsOption>`, I suggest to make TlsOption an required argument for servers module. It allows us setting more custom tls options, like:

- `disabled`: no cert/key configured, the server only provided plain connection
- `wanted`: cert/key configured, the server has secure connection but with plain one supported
- `required`: cert/key configured, the server requires secure connection and rejects clients using plain ones

 We can add these options in future patches but consider make TlsOption required with defaults to `disabled`.